test/src/actions/AppendMeshGeneratorAction.C

```diff
@@ -0,0 +1,43 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "AppendMeshGeneratorAction.h"
+#include "MooseMesh.h"
+#include "MeshGenerator.h"
+#include "Factory.h"
+#include "MooseApp.h"
+
+registerMooseAction("MooseApp", AppendMeshGeneratorAction, "append_mesh_generator");
+
+defineLegacyParams(AppendMeshGeneratorAction);
+
+InputParameters
+AppendMeshGeneratorAction::validParams()
+{
+  InputParameters params = MooseObjectAction::validParams();
+  return params;
+}
+
+AppendMeshGeneratorAction::AppendMeshGeneratorAction(InputParameters params)
+  : MooseObjectAction(params)
+{
+}
+
+void
+AppendMeshGeneratorAction::act()
+{
+  if (!_mesh)
+    mooseError("No mesh file was supplied and no generation block was provided");
+
+  if (!_moose_object_pars.have_parameter<MeshGeneratorName>("input"))
+    mooseError("Cannot append a mesh generator that does not take input mesh generators");
+
+  _moose_object_pars.set<MeshGeneratorName>("input") = _app.getCurrentFinalMeshGenerator();
+  _app.addMeshGenerator(_type, _name, _moose_object_pars);
```

Comment:
Is this your use case, or do you expect others? The reason I ask is I don't see a reason to expose the name of the last generator at all if your use case is to add to the "end". A better design would be to add a new method called "appendMeshGenerators" that just does the appending internally. Exposing the end as you've done in this PR has issues. As you've already stated, it doesn't quite work correctly if the user also explicitly sets the final generator.

Reply:
You are probably right. I do not need to expose this but rather create `appendMeshGenerator` function. Let me rewrite this PR.